expression/testdata/partition_pruner_in.json

```diff
@@ -0,0 +1,23 @@
+[
+  {
+    "name": "TestHashPartitionPruner",
+    "cases": [
+      // Point Select.
+      "explain select * from t1 where id = 7 and a = 6",
+      "explain select * from t2 where id = 9 and a = 1",
+      "explain select * from t2 where id = 9 and a = -110",
+      "explain select * from t1 where id = -17",
+      "explain select * from t2 where id = a and a = b and b = 2",
+      // Join.
+      "explain select * from t1 join t2 on (t1.id = t2.id) where t1.id = 5 and t2.a = 7",
+      "explain select * from t1 left join t2 on t1.id = 1 and t2.a = 2 where t2.id = 7",
+      "explain select * from t2 join t1 on t1.id = t2.id and t2.a = t1.id and t2.id = 12",
+      // Negtive cases.
+      "explain select * from t1 left join t2 on true where t1.a = 1 and false",
+      "explain select * from t1 left join t2 on true where t1.a = 1 and null",
+      "explain select * from t1 left join t2 on true where t1.a = null",
+      "explain select * from t1 where t1.a > 7 and t1.a < 3",
+      "explain select * from t1 where t1.a between 7 and 3"
```

Comment:
Add some complex ones like `a=b and b=c and c=d and d = e +1 and e=???` like ?

Reply:
This case may fail because `e + 1` will not be eval